### PR TITLE
Constant turn sandwich

### PR DESCRIPTION
**Pull request by snaylor20**
Adds a transition model which allows for a constant turn in 2 state variables which are not adjacent in the state vector, eg x and z in the (x, y, z) state vector

### CODE REVIEW

File: stonesoup/models/transition/linear.py [hunk 1]
```diff
@@ -744,74 +822,14 @@ class ConstantTurn(LinearGaussianTransitionModel, TimeVariantModel):
                       \end{bmatrix}
     """
 
-    noise_diff_coeffs = Property(
+    turn_noise_diff_coeffs = Property(
```
Comment (sdhiscocks) [inline stonesoup/models/transition/linear.py]: These properties are inherited, so no longer required here.

File: stonesoup/models/transition/tests/test_ct.py [hunk 2]
```diff
@@ -43,24 +44,24 @@ def base(model, state_vec, noise_diff_coeffs, turn_rate):
              [0, sp.sin(turn_ratedt),
               0, sp.cos(turn_ratedt)]])
 
-    qx = noise_diff_coeffs[0]
-    qy = noise_diff_coeffs[1]
-    Q = sp.array([[sp.power(qx, 2) * sp.power(timediff, 3) / 3,
```
Comment (etfrogers-dstl) [inline stonesoup/models/transition/tests/test_ct.py]: I assume `sp.array` will not merge well here, as `master` is now using `np.array`

This applies throughout

File: stonesoup/models/transition/linear.py [hunk 3]
```diff
@@ -501,7 +380,147 @@ def constant_derivative(self):
...
+        float, doc="The noise diffusion coefficient :math:`q`")
+    damping_coeff = Property(
+        float, doc="The Nth derivative damping coefficient :math:`K`")
+
+    @property
+    def ndim_state(self):
+        return self.decay_derivative + 1
+
+    @staticmethod
+    @lru_cache()
```
Comment (etfrogers-dstl) [inline stonesoup/models/transition/linear.py]: Like the use of `lrucache`. Have you considered whether an explicit cache size is useful? Is the default of 128 sensible given the likely number of combinations of `N` and `K`?

File: stonesoup/models/transition/tests/test_ct.py (hunk 2 above)
Comment (sdhiscocks) [reply stonesoup/models/transition/tests/test_ct.py]: I can take care of this when merging.

**Pull request merged**